src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -636,6 +638,25 @@ public void unregisterVerticleFactory(VerticleFactory factory) {
   }
 
   @Override
+  public Vertx addSchedulerInterceptor(BiFunction<Context, Runnable, Runnable> interceptor) {
+    schedulerInteceptors.add(interceptor);
+    return this;
+  }
+
+  @Override
+  public Vertx removeSchedulerInterceptor(BiFunction<Context, Runnable, Runnable> interceptor) {
+    schedulerInteceptors.remove(interceptor);
+    return this;
+  }
+
+  @Override
+  public Runnable interceptScheduledWork(final ContextImpl context, final Runnable task) {
+    return schedulerInteceptors.stream()
+        .reduce((c, t) -> t, (a, b) -> (c, t) -> b.apply(c, a.apply(c, t)))
```

Comment:
I found this line elegant but also confusing to read. Maybe a more imperative version would improve readability (and also improve perf a bit).